### PR TITLE
:sparkles: Writer XML

### LINKED ISSUE #5: 🖨️ Writing XML
# 👮‍♂️ What's needed
🖨️ Writing the content of the data struct for each type of elements to a XML file

## 📋 Subtasks
- [x] Write the element of the data structure in a XML file 
- [x] Prettier the XML into the good format with correct indentation  

# 📜 Sources
No specific sources

exemple of xml file

```xml
<document>
    <header title="Syntaxe XML">
        <author>Fornes Leo</author>
        <date>2024-01-01</date>
    </header>
    <body>
        <paragraph>This document is a simple example of the XML syntax.</paragraph>
        <paragraph>Every syntax element is displayed in this document.</paragraph>
        <section title="header 1">
            <paragraph>This is a basic paragraph with text.</paragraph>
            <paragraph>This is a paragraph with <bold>bold</bold>, <italic>italic</italic> and <code>code</code> text.</paragraph>
            <section title="header 2">
                <codeblock>
                    <paragraph>This is a code block.</paragraph>
                </codeblock>
                <list>
                    <paragraph>list item 1</paragraph>
                    <paragraph>list item 2</paragraph>
                    <paragraph>list item 3</paragraph>
                </list>
                <paragraph>This is a paragraph with a <link url="https://www.youtube.com/watch?v=dQw4w9WgXcQ&ab_channel=RickAstley">link</link>.</paragraph>
                <paragraph>This is a paragraph with an image<image url="https://cdn-images-1.medium.com/max/697/1*tsHrUKwQXG1YZX0l957ISw.png">Text to replace image</image>.</paragraph>
                <section title="">
                    <section title="header 4">
                        <paragraph>Every syntax element can be use separately or combined.</paragraph>
                        <paragraph>Think about the different possible combinations.</paragraph>
                        <paragraph>All combined syntax elements aren't displayed in this document.</paragraph>
                    </section>
                </section>
            </section>
        </section>
    </body>
</document>
```

### PR DESCRIPTION
**Pull request by landryarki**
added function documentToXml inside module WriterXml
```hs
documentToXml :: Document -> String
```

i made some minor fixes about indentation inside the parser that i noticed only when i printed the reconverted file

### CODE REVIEW

**Review by 6im0n — APPROVED** [review]
Good for me !

File: src/WriterXml.hs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+module WriterXml (
+        writeXmlDocument
+    ) where
+      
+import DataStruct
+import Prelude
+
+writeXmlDocument :: FilePath -> Document -> IO ()
+writeXmlDocument path doc = writeFile path (documentToXml doc)
```
Comment (6im0n) [inline src/WriterXml.hs]: send only a string of the parsed file is ok ;)

**Review by 6im0n — CHANGES_REQUESTED** [review]

**Review by 6im0n — APPROVED** [review]
Good for me ! Thanks for the change ! :+1:

**Review by axel-eck — APPROVED** [review]
Nice, just if you don't mind to move your parser and writer in a sub-folder named 'xml' so each pair of writer/parser is in its own package

**Pull request merged**